### ISSUE TITLE
Standardize and Publish ZeroNet Protocol Specifications.

### ISSUE THREAD

**Issue by canewsin** [HelloZeroNet/ZeroNet#2425]
Since there is no documentation on ZeroNet Protocol Specifications, it is better to start standardize the low level specifications, so it's easy to implement on various other languages, see for example BitTorrent has various beps specific to features, I was considering ZeroNet as next(third) generation internet and moving with usages of this type of interest, I already drawn up problems faced by these type of systems and solutions for them on table. Also if these specifications are standardized, then we can implement this in lower level languages for performance benifits that improves the ZeroNet Network.

**Comment by russianagent** [comment]
Forget it @canewsin ZeroNet don't even have a valid enforcable license! The current GPLv2 license is terminated! If the license is not replaced with GPLv3+ ZeroNet is a dead project. I personally using a GPLv3+ license on my local copy of ZeroNet. Don't rush with any standard here. Only 3 person active really @imachug @shortcutme and the crazy @filips123 and me of course... Sometimes I see around here @rllola too. Other people since the license is terminated simply don't give a damn about ZeroNet.. 

License issue what I reported and triggered a voting which was manipulated by @imachug  https://github.com/HelloZeroNet/ZeroNet/issues/2273 

So until the license is not replaced consider ZeroNet as a big trash and illegal. If you copy and you distribute it under a GPLv2 license you can be even jailed in some countries!!!!

**Comment by filips123** [comment]
@canewsin Don't worry, russianagent is just spammer/troll which has been spamming (and making death threats) for a few months.

Also, some documentation of ZeroNet protocol is [here](https://zeronet.io/docs/help_zeronet/network_protocol/). But more standardization would be good.

**Comment by russianagent** [comment]
@filips123 you live in dream world until not kicking your door down and show you the harsh environment outside in the real world! until the license in not replaced ZeroNet will be considered just a big shitpile which not even worth mentioning anywhere and this is thanks to you my dear retarded friend.

**Comment by russianagent** [comment]
Tell me what license ZeroNet has right now? I tell you! **NOTHING!**

**Comment by russianagent** [comment]
So @canewsin it is not worth to spend your time on ZeroNet or in any way help them. I contributed to this repository various times and because I was dare to report that ZeroNet illegally distributed more than 20 account of mine is suspended by GitHub. 

And they claim censorship resistance... Big joke @HelloZeroNet !

**Comment by filips123** [comment]
> I was dare to report that ZeroNet illegally distributed more than 20 account of mine is suspended by GitHub.

Maybe your account was suspended because you were making death threats, which are million times more illegal than any license violation?

**Comment by russianagent** [comment]
Prove it that I made death threats! You can't because I don't... I'm an angel :angel:

**Comment by russianagent** [comment]
Anyway, regardless of your opinion ZeroNet doesn't deserve any contribution until the license is not fixed! **Takes less than 10 second to fix the license** but @HelloZeroNet is a lazy bitch. :rofl: 

So yeah, no need standards until that....

**Comment by russianagent** [comment]
I like you @filips123 just that you know that. I like @imachug and @shortcutme too but seriously guys need to fix that license!!!!!

**Comment by filips123** [comment]
Look. I would also like to fix the license.

But you can't simply replace the old file with new one. You need permission from copyright holders (contributors) to do this, and not all of them allowed this.

**Comment by canewsin** [comment]
@filips123 I also agree with openness of the project because more open the project, more speed development will catch up but at the end of the it's people choice who write the code,
@russianagent 
One thing to note is I like ZeroNet, if the code is licensed far from open world I write my own code or copy paste code samples of others found on internet if I don't know what to do, but at this time I don't want to take from scratch, I also planned some ambitious project on ZeroNet, so I don't want to step back from this, since I always works around pluggable api, felt your copy suits me I will use your @russianagent repo I don't bother either way, also please read issue carefully because even if you write from scratch you may use same protocol on your copy, that is what I was asking.

**Comment by russianagent** [comment]
@filips123 Copyright permission is granted already under GPL as @shortcutme explained 90+% of the contributors voted for GPLv3+. You as an individual before distributing ZeroNet must replace the license with GPL-3.0 or preferably GPL-3.0-or-later. 

I noted that GIMP uses GPLv3+ and today I tell you that Wire (https://github.com/wireapp) also uses GPLv3+ on all of its clients including desktop and android... 

The greatest open source software use GPLv3 and I beleive ZeroNet should grow up and we could add ZeroNet to https://savannah.nongnu.org too, that would be a real boost! 

I love ZeroNet because it is a great software and I will always use it because I think ZeroNet will be the **STANDARD** for all websites today you see online! Tamas did the right thing when started ZeroNet I admire him for this!  

@canewsin I have made significant changes in ZeroNet because I serving my websites with it. Front-end is a reverse proxy securely to the back-end where ZeroNet is running in full Tor mode. I see enormous potential in ZeroNet and what I'm doing maybe the greatest use case of ZeroNet in its history. I will not disclose what I'm doing with ZeroNet but one thing is for sure I'm not the guy who spreading Nazi trash like 8chan...

**Comment by filips123** [comment]
> Copyright permission is granted already under GPL 

Permission was only granted under GPLv2. If it would be granted under GPLv2-or-later, you could easily update the license, but as it is under GPLv2-only, you can't.

**Comment by russianagent** [comment]
Look, I already explained this. The GPLv2 license is GONE! Permanently terminated and no matter what you do that you can't get it back! Same with all forks!!!! So if it is like this I'm legally obligated to use a license which is GPL and that is GPLv3(+)!! I know you don't like it but this is the only way how forks and you can distribute ZeroNet legally!

**Comment by filips123** [comment]
So if "GPLv2 license is GONE," ZeroNet is now proprietary, so you also need permission from contributors to change license...

**Comment by russianagent** [comment]
No because contributors already voted for GPLv3+ https://github.com/HelloZeroNet/ZeroNet/issues/2273!!!! They made a statement publicly on the recent voting which gives you and everyone else explicit permission to do the replacement even if this repository is not replaced the terminated license!

**Comment by filips123** [comment]
*All* contributors need to agree...

**Comment by russianagent** [comment]
Some contributors are assholes and some are just simply inactive. Active contributors did agree!
